src/bootstrap/src/core/config/config.rs

```diff
@@ -776,6 +776,30 @@ impl Merge for TomlConfig {
             }
         }
 
+        self.change_id.inner.merge(None, &mut Default::default(), change_id.inner, replace);
+        self.profile.merge(None, &mut Default::default(), profile, replace);
```

Comment:
I was confused by this at first, but then I realized that `profile` merging is "lazy"; in includes, only the `profile` value is overwritten, but the final profile is only applied in the root config, not for the intermediate included configs. I guess that in most cases this won't be an issue, but it deserves at least a comment.

Reply:
I don't get it, could you clarify how does this cause an override? It should only use the root profile and ignore the rest if there are any.